.changeset/moody-waves-think.md

```diff
@@ -0,0 +1,9 @@
+---
+'astro': major
+---
+
+Updates the Astro config loading flow to ignore processing locally-linked dependencies with Vite (e.g. `npm link`, in a monorepo, etc). Instead, they will be normally imported by the Node.js runtime the same way as other dependencies from `node_modules`.
+
+Previously, Astro would process locally-linked dependencies, which they were able to use Vite features like TypeScript when imported by the Astro config file. However, it also caused confusion as integration authors may test against a package that worked locally, but not when published. It also restricts using CJS-only dependencies as Vite requires the code to be ESM. Hence, the behaviour is now changed to ignore processing any type of dependencies by Vite.
```

Comment:
```suggestion
Previously, Astro would process locally-linked dependencies when they were able to use Vite features like TypeScript when imported by the Astro config file. 

However, this caused confusion as integration authors may test against a package that worked locally, but not when published. This method also restricts using CJS-only dependencies because Vite requires the code to be ESM. Therefore, Astro's behaviour is now changed to ignore processing any type of dependencies by Vite.
```
should that be a "when they were able to use"? "which were able to use"?

Reply:
Oh, yeah "which were able to use" would be more accurate and concise. It's the locally-linked dependencies that can use the Vite features previously and now they can't